diff --git a/extensions/json-language-features/server/src/jsonServer.ts b/extensions/json-language-features/server/src/jsonServer.ts
--- a/extensions/json-language-features/server/src/jsonServer.ts
+++ b/extensions/json-language-features/server/src/jsonServer.ts
@@ -17,0 +18 @@
+import requestLight from 'request-light';
@@ -147,0 +149,2 @@
+		setSchemaStoreSettingsIfNotSet();
+
@@ -245,0 +249 @@
+	let schemaStoreSettings: SchemaConfiguration[] = [];
@@ -294,0 +299,34 @@
+
+	function setSchemaStoreSettingsIfNotSet() {
+		if (schemaStoreSettings.length > 0) {
+			return;
+		}
+		getSchemaStoreMatchingSchemas().then(schemas => {
+			schemaStoreSettings = schemas;
+			updateConfiguration();
+		}, error => {
+			connection.console.error(`Unable to fetch SchemaStore catalog: ${error}`);
+		});
+	}
+
+	function getSchemaStoreMatchingSchemas(): Promise<SchemaConfiguration[]> {
+		return requestLight.xhr({
+			url: 'https://www.schemastore.org/api/json/catalog.json',
+			followRedirects: 5
+		}).then(response => {
+			const catalog = JSON.parse(response.responseText);
+			const schemas = new Array<SchemaConfiguration>();
+
+			for (const schema of catalog.schemas) {
+				schemas.push({
+					uri: schema.url,
+					fileMatch: schema.fileMatch || []
+				});
+			}
+
+			return schemas;
+		}, error => {
+			throw error;
+		});
+	}
+
@@ -393,0 +431,3 @@
+		if (schemaStoreSettings.length > 0) {
+			languageSettings.schemas.push(...schemaStoreSettings);
+		}
